h/services/annotation_delete.py

```diff
@@ -19,6 +19,14 @@ def delete(self, annotation):
         """
         annotation.updated = datetime.utcnow()
         annotation.deleted = True
+        self.request.db.add(
+            Job(
+                name="delete_annotation",
+                priority=0,
+                tag="h.services.annotation_delete.delete",
+                kwargs={"annotation_id": annotation.id},
+            )
+        )
```

Comment:
When deleting an annotation add a `delete_annotation` job to the queue in the same transaction, so that the annotation definitely gets deleted from Elasticseatch.

(This isn't actually part of the user delete feature, it's just for the normal annotation delete API, but it's something I think I may as well fix.)